### ISSUE TITLE
the meaning of RetinaFace 5 facepoint

### ISSUE THREAD

**Issue by xiakj** [deepinsight/insightface#889]
# 1--Handshaking/1_Handshaking_Handshaking_1_292.jpg
612 111 96 125 635.929 163.571 1.0 686.214 158.857 1.0 663.429 189.5 0.0 646.929 203.643 0.0 681.5 199.714 0.0 0.77
709 105 60 112 749.0 153.0 0.0 763.0 148.0 1.0 765.0 173.0 0.0 745.0 196.0 0.0 757.0 193.0 0.0 0.7
313 193 71 89 353.062 211.125 0.0 376.688 207.75 0.0 379.5 234.188 0.0 354.75 255.562 0.0 375.562 252.188 0.0 0.66
177 218 64 109 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 -1.0 0.68

in the retinaface paper, the author describe "In total, we have annotated 84.5k faces on the training set and 18.5k faces on the validation set", and I download the label from [https://pan.baidu.com/s/1Laby0EctfuJGgGMgRRgykA](url), the layout seems as "x1 y1 w h p1_x p1_y p1_conf p2_x p2_y p2_conf...". But there are three confidence? What's the meaning of "-1.0 0.0 1.0"?

**Comment by israelwei** [comment]
Based on the comments in the code and inspecting the faces in images with these kind of confidences, I think that:
-1   --- > missing landmark (not annotated)
1   ---> non-visible landmark. This will be the landmark's value due to occlusions etc. For example in sunglasses the eyes' landmarks confidence will be 1. Also for profile faces if only one eye is visible, the other eye's confidence will be 1. 
0 ---> visible landmark

**Comment by LI-ZONG-HAN** [comment]
Dose anyone knows the meaning of the right most value for each rows?
ex. 612 111 96 125 635.929 163.571 1.0 686.214 158.857 1.0 663.429 189.5 0.0 646.929 203.643 0.0 681.5 199.714 0.0 0.77

what 0.77 stands for

**Comment by xiakejiang** [comment]
> Dose anyone knows the meaning of the right most value for each rows?
> ex. 612 111 96 125 635.929 163.571 1.0 686.214 158.857 1.0 663.429 189.5 0.0 646.929 203.643 0.0 681.5 199.714 0.0 0.77
> 
> what 0.77 stands for

I guess the meaning of the right most value for each rows is the "face image quality", you can check it at the paper Figure 4.

**Comment by clearlove7-piggy** [comment]
> > Dose anyone knows the meaning of the right most value for each rows?
> > ex. 612 111 96 125 635.929 163.571 1.0 686.214 158.857 1.0 663.429 189.5 0.0 646.929 203.643 0.0 681.5 199.714 0.0 0.77
> > what 0.77 stands for
> 
> I guess the meaning of the right most value for each rows is the "face image quality", you can check it at the paper Figure 4.

how  to set the confidence score like 0.77? set it randomly?

**Comment by xiakj** [comment]
@clearlove7-piggy Ha, that's a good question!
